src/page-weekly-snapshot/projectionChartUtils.tsx

```diff
@@ -0,0 +1,187 @@
+import { LocaleData } from "../locale-data-context";
+import { ModelInputs } from "../page-multi-facility/types";
+import { combineFacilitiesProjectionData } from "../page-response-impact/responseChartData";
+
+export function getCaseCount(curveData: ndarray<number>) {
+  const cases = [];
+  for (let day = 0; day < curveData.shape[0]; ++day) {
+    const caseCountForDay = countCasesForDay(curveData, day);
+    cases.push(caseCountForDay);
+  }
+  return cases;
+}
+
+export const today = dateFns.endOfToday();
+export const ninetyOneDaysAgo = dateFns.subDays(today, 91);
+export const ninetyDaysAgo = dateFns.subDays(today, 90);
+
+function addProjectionPadding(values: number[], numDays = 90): number[] {
+  return [...Array(numDays - values.length).fill(0), ...values];
+}
+
+export function getVersionForProjection(versions: ModelInputs[]) {
+  const versionsWithCases = versions
+    .sort(({ observedAt: a }, { observedAt: b }) => dateFns.compareAsc(a, b))
+    .filter((version) => totalConfirmedCases(version) > 0);
+
+  const version90DaysAgo = versionsWithCases.find((v) =>
+    dateFns.isSameDay(v.observedAt, ninetyDaysAgo),
+  );
+
+  let versionForProjection: ModelInputs | undefined = version90DaysAgo;
+
+  // If the version from 90 days ago has 0 or no data, look forward for the next
+  // version with cases
+  if (!version90DaysAgo) {
+    const closestVersionWithCases = versionsWithCases.find((version) => {
+      const date = dateFns.closestTo(
+        ninetyDaysAgo,
+        versionsWithCases.map((v) => v.observedAt),
+      );
+      return dateFns.isSameDay(version.observedAt, date);
+    });
+    versionForProjection = closestVersionWithCases;
+  }
+  return versionForProjection;
+}
+
+export function getProjectedData(epidemicModelInputs: EpidemicModelInputs) {
+  if (!epidemicModelInputs || !epidemicModelInputs?.observedAt) {
+    throw `No version found with cases`;
+  }
+
+  let numDaysAfterNinetyDays = 0;
+
+  if (dateFns.isAfter(epidemicModelInputs.observedAt, ninetyOneDaysAgo)) {
+    numDaysAfterNinetyDays = dateFns.differenceInCalendarDays(
+      epidemicModelInputs.observedAt,
+      ninetyOneDaysAgo,
+    );
+  }
+
+  const projectionNumDays = NUM_DAYS - numDaysAfterNinetyDays;
+  const curveInputs = curveInputsFromUserInputs(epidemicModelInputs);
+  const curveData = calculateCurves(curveInputs, projectionNumDays);
+
+  if (!curveData) {
+    return {
+      projectedFatalities: addProjectionPadding([]),
+      projectedCases: addProjectionPadding([]),
+    };
+  }
+
+  const summedProjectionData = combineFacilitiesProjectionData(
+    [curveData],
+    projectionNumDays,
+  );
+
+  return {
+    projectedFatalities: addProjectionPadding(
+      getAllValues(getColView(summedProjectionData, seirIndex.fatalities)),
+    ),
+    projectedCases: addProjectionPadding(getCaseCount(summedProjectionData)),
+  };
+}
+
+export function getActualDataForFacility(modelVersions: ModelInputs[]) {
```

Comment:
A comment: logic in generic files like `*Utils` should tend to have docstrings since 1) it tends to be where we put dense logic to spare callers, and 2) it tends to be general-purpose for all kinds of varied consumers, which makes it important for future developers to easily understand the functionality.

Reply:
Yes that makes sense...ideally I would want to add tests for these too. I added comments, and it's making me see that maybe some of these functions are doing too much. Let me know what you think, and if there is a different format that would be better for the comments.

Reply:
These look perfect. Thanks!